config/apply/ccsm/translate.js

```diff
@@ -435,53 +424,59 @@ export function translateResponse(customApiResponse, patientData) {
       conclusionCodes = mapResults(excisionResults, customExcisionCodes, standardHistologyCodes, conclusionCodes);
     }
 
-    if (diagnosticReportIndex !== -1) {
-      console.log('Found the diagnostic report reference');
-      // Update the DiagnosticReport in patientData with the results from the custom API
-      patientData[diagnosticReportIndex] = {
-        ...patientData[diagnosticReportIndex],
-        code: {
-          coding: [
-            ...patientData[diagnosticReportIndex].code.coding,
-            ...codings
-          ]
-        },
-        conclusionCode: conclusionCodes
-      };
-
-      console.log('diagnostic report: ', patientData[diagnosticReportIndex]);
-      console.log('dr conclusion codes: ', patientData[diagnosticReportIndex].conclusionCode);
-      patientData[diagnosticReportIndex].conclusionCode.forEach(drcc => {
-        console.log('dr mapped conconclusion code: ', drcc.coding[0], drcc.text);
-      });
+    let patient = patientData.find(pd => pd.resourceType === 'Patient');
 
-      // Create a Procedure resource based on DiagnosticReport
-      if (procedureCoding) {
-        const originalDiagnosticReport = patientData[diagnosticReportIndex];
-        let newProcedure =
+    const laboratoryCategoryCode = {
+      'coding': [
         {
-          'resourceType': 'Procedure',
-          'id': originalDiagnosticReport.id,
-          'subject': originalDiagnosticReport.subject,
-          'status': 'completed',
-          'code': {
-            'coding': [procedureCoding],
-            'text': procedureText
-          },
-          'performedDateTime': originalDiagnosticReport.effectiveDateTime
-        };
-
-        if (newProcedure.id.length > 54) {
-          newProcedure.id = newProcedure.id.substring(0, 54) + '-procedure';
-        } else {
-          newProcedure.id += '-procedure';
+          'system': 'http://terminology.hl7.org/CodeSystem/v2-0074',
+          'code': 'LAB',
+          'display': 'Laboratory'
         }
+      ],
+      'text': 'Laboratory'
+    };
 
-        patientData.push(newProcedure);
+    // Create a DiagnosticReport resource from Order
+    let newDiagnosticReport = {
```

Comment:
I outputted a DiagnosticReport that was created from running the tests, and ran the FHIR Validator on it against the [US Core DiagnosticReport Profile for Laboratory Results Reporting](https://build.fhir.org/ig/HL7/US-Core/StructureDefinition-us-core-diagnosticreport-lab.html). The validator showed the following error: 

> Constraint failed: us-core-9: 'issued SHALL be present if the status is 'partial', 'preliminary', 'final', 'amended', 'corrected' or 'appended''

While this isn't critical for the DiagnosticReport to work with our CDS, I think we should add the order.ResultDate to the `DiagnosticReport.issued` field as well, since there are other places where we expect resource instances to align with U.S. Core.

Reply:
This was addressed in commit https://github.com/ccsm-cds-tools/ccsm-cds-service/pull/27/commits/1e99515f3c50f01dc13a1a5c4342f2abcaa507d2. Thanks!